backend-lib/src/main/rust/lib.rs

```diff
@@ -564,7 +564,10 @@ pub extern "C" fn Java_cash_z_ecc_android_sdk_internal_jni_RustBackend_importAcc
                 } else {
                     None
                 };
-                let hd_account_index = zip32::AccountId::try_from(hd_account_index_raw).ok();
+
+                let hd_account_index_checked = hd_account_index_raw as u32;
+
+                let hd_account_index = zip32::AccountId::try_from(hd_account_index_checked).ok();
```

Comment:
This is a bit difficult to reason about. I _think_ it might be fine (because `-1u64` might be cast to `0xffffffff` which would be outside the valid range for `zip32::AccountId`), but it is easier to separate these two:
```suggestion
                let hd_account_index = u32::try_from(hd_account_index_raw)
                    .ok()
                    .and_then(|hd_account_index_non_null| zip32::AccountId::try_from(hd_account_index_non_null).ok());
```